pkg/agent/apiserver/apiserver.go

```diff
@@ -124,10 +125,10 @@ func newConfig(bindPort int, enableMetrics bool, kubeconfig string) (*genericapi
 	// Set the PairName but leave certificate directory blank to generate in-memory by default.
 	secureServing.ServerCert.CertDirectory = ""
 	secureServing.ServerCert.PairName = Name
-	secureServing.BindAddress = net.ParseIP("0.0.0.0")
+	secureServing.BindAddress = net.ParseIP(ip.UnspecifiedMap[ipFamily])
 	secureServing.BindPort = bindPort
 
-	if err := secureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
+	if err := secureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP(ip.LocalhostMap[ipFamily])}); err != nil {
```

Comment:
I'm not 100% sure, but it feels like we could unconditionally add both localhost addresses (v4 & v6) in these certificates

Reply:
I think so. I have tested part of e2e tests locally and they were successful. I will trigger all CI tests to verify.